packages/ra-ui-materialui/src/list/filter/SavedQueriesList.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import { ReactNode } from 'react';
+import { Typography, List, Stack, styled, Tooltip } from '@mui/material';
+import BookmarkIcon from '@mui/icons-material/BookmarkBorder';
+import HelpIcon from '@mui/icons-material/HelpOutline';
+import { useListContext, useTranslate } from 'ra-core';
+import isEqual from 'lodash/isEqual';
+
+import { useSavedQueries } from './useSavedQueries';
+import { RemoveSavedQueryIconButton } from './RemoveSavedQueryIconButton';
+import { AddSavedQueryIconButton } from './AddSavedQueryIconButton';
+import SavedQueryFilterListItem from './SavedQueryFilterListItem';
+
+/**
+ * FilterList-like component allowing to save and restore a query (filters, sort, perPage).
+ *
+ * Use this component in a Filter sidebar to let users store custom queries
+ * that they can reuse later. The saved queries will appear as FilterListItems,
+ * and can be removed.
+ *
+ * This component uses useStore under the hood to store saved queries in
+ * localStorage, one set of saved queries per resource.
+ *
+ * @example
+ *
+ * import { FilterList, FilterListItem, List, Datagrid, SavedQueriesList } from 'react-admin';
+ * import { Card, CardContent } from '@mui/material';
+ *
+ * const PostFilterSidebar = () => (
+ *     <Card>
+ *         <CardContent>
+ *             <SavedQueriesList />
+ *             <FilterList label="Category" icon={<CategoryIcon />}>
+ *                 ...
+ *             </FilterList>
+ *             <FilterList label="Published" icon={<DateRangeIcon />}>
+ *                ...
+ *             </FilterList>
+ *             <FilterList label="Popularity" icon={<DateRangeIcon />}>
+ *                ...
+ *             </FilterList>
+ *         </CardContent>
+ *     </Card>
+ * );
+ *
+ * const PostList = () => (
+ *     <List aside={<PostFilterSidebar />}>
+ *         <Datagrid>
+ *             ...
+ *         </Datagrid>
+ *     </List>
+ * );
+ *
+ */
+export const SavedQueriesList = ({
+    icon = defaultIcon,
+}: SavedQueriesListProps) => {
+    const translate = useTranslate();
+    const {
+        resource,
+        filterValues,
+        displayedFilters,
+        sort,
+        perPage,
+    } = useListContext();
+
+    const [savedQueries] = useSavedQueries(resource);
+
+    const hasSavedCurrentFilterValue = savedQueries.some(savedQuery =>
+        isEqual(savedQuery.value, {
+            filter: filterValues,
+            sort,
+            perPage,
+            displayedFilters,
+        })
+    );
+    const hasFilterValues = !isEqual(filterValues, {});
+
+    // note: we don't use react-admin's FilterList because it doesn't accept a default translation
```

Comment:
I don't understand this one

Reply:
Wrongly copy/pasted from ra-preferences. Refactored